whois-api/src/main/java/net/ripe/db/whois/api/rest/ReferencesService.java

```diff
@@ -187,21 +187,18 @@ public Response create(
 
             validateObjectNotFound(whoisResources, mntner);
 
-            final WhoisResources updatedResources = performUpdates(request, actionRequests, passwords, crowdTokenKey, null, SsoAuthForm.ACCOUNT, null);
+            final Origin origin = updatePerformer.createOrigin(request);
+            final UpdateContext updateContext = updatePerformer.initContext(origin, crowdTokenKey, request);
+            updateContext.setBatchUpdate();
+            updateContext.setBasicAuth(BasicAuthExtractor.getBasicAuth(request));
+            auditlogRequest(request);
+
+            final List<Update> updates = prepareUpdates(updateContext, actionRequests, passwords, null, SsoAuthForm.ACCOUNT, null);
+            final WhoisResources updatedResources = performUpdates(origin, updateContext, updates, request);
```

Comment:
We were adding in one transaction somethings not related to the call to the database. 
I split this big method in two, one of them to prepareUpdates and the other one "performUpdates" just to apply the calls to the database in the transaction